### PR TITLE
Code cleanup - take 2

### PR DESCRIPTION
**Pull request by henkmollema**
Utilize C# features:
- Expression bodies
- Auto-properties (also get-only properties)
- Null propagation
- `nameof` operator
- String interpolation
- Dictionary initializer
- await in finally clause

### CODE REVIEW

**Comment by henkmollema** [comment]
I'll add a code clean-up commit later.

File: Dapper/SqlMapper.GridReader.Async.cs [hunk 1]
```diff
@@ -104,27 +104,23 @@ private Task<IEnumerable<T>> ReadAsyncImpl<T>(Type type, bool buffered)
...
+                    var reader = (DbDataReader)this.reader;
+                    List<T> buffer = new List<T>();
+                    while (index == gridIndex && await reader.ReadAsync(cancel).ConfigureAwait(false))
+                    {
+                        buffer.Add((T)deserializer(reader));
+                    }
+                    return buffer;
                 }
-                if (index == gridIndex) // need to do this outside of the finally pre-C#6
+                finally // finally so that First etc progresses things even when multiple rows
```
Comment (henkmollema) [inline Dapper/SqlMapper.GridReader.Async.cs]: @mgravell is this the right approach here?

Comment (mgravell) [inline Dapper/SqlMapper.GridReader.Async.cs]: Yes, I think that's right; IIRC that was the code we had originally, targeting C# 6 (the language version no longer being so much of a pain point)

**Comment by henkmollema** [comment]
Added code-clean up as well.

/cc @NickCraver and @mgravell

**Comment by NickCraver** [comment]
@henkmollema I'm ready to merge the first commit - on the 2nd though I think it's best we revert and squash down. It's mostly coding style like braces/alignment/etc. and I think both Marc and I prefer a much more terse approach when it's clear what's happening. There's some things in there I want to address, but I think it'll be far easier for me to do a quick cleanup on this side after merging the C#6 changes in. Can you please peel off the 2nd commit (revert and squash is probably easiest).

And don't worry about the Concat thing - it'll get love on the side anyway.

**Comment by henkmollema** [comment]
Sure thing. I'll add some more C# 6 to the first commit and squash it into one, without the code clean-up.

**Comment by henkmollema** [comment]
@NickCraver updated.

On a side note: how do you comment on a commit ([this](https://github.com/henkmollema/dapper-dot-net/commit/0b8cf92ba22eb03beef431d291e4671234c097cf#commitcomment-14328627) and [this](https://github.com/henkmollema/dapper-dot-net/commit/0b8cf92ba22eb03beef431d291e4671234c097cf#commitcomment-14328654))? Because they are not showing up in this PR, just as comment on the commit. Makes it hard to keep track of the feedback. :wink:

**Comment by NickCraver** [comment]
Oh you just click the plus on the line number to leave one - don't worry if they get lost in rebases :)

**Comment by henkmollema** [comment]
Yeah I know. But somehow yours did not show up here. I had to check my email.

**Comment by NickCraver** [comment]
Merging in - thanks!

**Comment by henkmollema** [comment]
Awesome! :smile:

**Pull request merged**